### PR TITLE
Initialize directory for production installation

### LINKED ISSUE #370: Create VMs for production installation and document the process
This is already done, creating the ticket for better tracking and to have something for the PR to reference to.

### PR DESCRIPTION
**Pull request by Erethon**
This PR initializes the directory for the production installation of the security tracker. It currently:
- Creates a new directory and a simple README for the infra
- Initializes a Terraform/OpenTofu setup for our Hetzner Cloud VMs
 
Closes #370

### CODE REVIEW

File: infra/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Running `nix shell` on the `terraform` directory will drop you in a shell with
+OpenTofu available. To quickly get information on the committed state run `tofu
+show`. For more instructions on how to use OpenTofu refer to the [upstream documentation](https://opentofu.org/docs/).
+
+## NixOS hosts
+
+Since Hetzner Cloud doesn't support NixOS out of the box, the VM was initially
+spawned as a Debian host and then it was converted to NixOS as per the
+[instructions on the NixOS wiki](https://wiki.nixos.org/wiki/Install_NixOS_on_Hetzner_Cloud#Traditional_ISO_installation).
```
Comment (fricklerhandwerk) [inline infra/README.md:23]:
```suggestion
[provisioning NixOS via SSH tutorial](https://nix.dev/tutorials/nixos/provisioning-remote-machines).
```

**Review by fricklerhandwerk — CHANGES_REQUESTED** [review]
- `infra/shell.nix` is missing
- please add URLs to where to manage the Hetzner deployment, and where to find out who has access

**Review by fricklerhandwerk — APPROVED** [review]

**Comment by Erethon** [comment]
I just noticed the changes pushed in this PR. @fricklerhandwerk is there a reason to move the terraform files outside of the terraform directory?

Moreover, since the move, `nix-shell` doesn't work because we're pointing to the wrong npins (two directories up instead of one).  https://github.com/erethon/nix-security-tracker/commit/50da9709310d8a5002a5e619bf1ad6ecc2a27512 fixes this, but I'm not a fan of using the versions of this specific npins file. I wonder what's the proper way to specify a particular branch (say nixos-24.11) just for this shell.

**Comment by fricklerhandwerk** [comment]
Ah right, I was a bit fast, sorry. You can just initialise a separate `npins` in that subdirectory and import from there. One directory for one file seemed a bit excessive, I took the liberty excercise my authority as product owner to keep code readability in check for myself. :) If that was naive of me, feel free to change it back with a line of rationale.

**Pull request merged**